src/udev/udev-rules.c

```diff
@@ -1170,6 +1189,10 @@ int udev_rule_parse_value(char *str, char **ret_value, char **ret_endpos) {
                 str[l + 1] = '\0';
         }
 
+        /* convert to lowercase after string is escaped */
+        if (to_lowercase)
+                ascii_strlower(str);
```

Comment:
I do not think `"i"` should make lower case the input string, but string comparison should be done in case insensitive.

Reply:
Agree, after sleeping on this, I think its better to make `==i` or `!=i` do case-insensitive comparison. Alternatively you could add another operator like `~=`

However, looks like you already did this in https://github.com/systemd/systemd/pull/34425, thanks!